src/toast/ops/simple_deglitch.py

```diff
@@ -0,0 +1,245 @@
+                        # Special treatment for the ends
+                        sig_view[:w] -= np.median(sig_view[:w])
+                        sig_view[-w:] -= np.median(sig_view[-w:])
+                    trend = sig[ind] - sig_view
+                    sig_view[bad[ind]] = np.nan
+                    offset = np.nanmedian(sig_view)
+                    sig_view -= offset
+                    trend += offset
+                    rms = np.nanstd(sig_view)
+                    nglitch = 0
+                    while True:
+                        if np.isnan(rms) or \
+                           np.sum(np.isfinite(sig_view)) < self.nsample_min:
+                            # flag the entire view.  Not enough statistics
+                            sig_view[:] = np.nan
+                            break
+                        # See if the brightest remaining sample still stands out
+                        i = np.nanargmax(np.abs(sig_view))
+                        sig_view_test = sig_view.copy()
+                        istart = max(0, i - self.glitch_radius)
+                        istop = min(nsample, i + self.glitch_radius + 1)
+                        sig_view_test[istart : istop] = np.nan
+                        rms_test = np.nanstd(sig_view_test)
+                        if np.abs(sig_view[i]) < self.glitch_limit * rms_test:
+                            # Not significant enough
+                            break
+                        nglitch += 1
+                        sig_view = sig_view_test
+                        rms = rms_test
+                    if nglitch == 0:
+                        continue
+                    bad_view = np.isnan(sig_view)
+                    det_flags[ind][bad_view] |= self.glitch_mask
+                    if self.fill_gaps:
+                        nbad = np.sum(bad_view)
+                        corrected_signal = sig[ind].copy()
+                        corrected_signal[bad_view] = trend[bad_view]
+                        corrected_signal[bad_view] += np.random.randn(nbad) * rms
+                        # DEBUG begin
+                        # import pdb
+                        # import matplotlib.pyplot as plt
+                        # pdb.set_trace()
+                        # DEBUG end
+                        sig[ind] = corrected_signal
+
+        return
+
+    def _finalize(self, data, **kwargs):
+        return
+
+    def _requires(self):
+        req = {
+            "meta": list(),
+            "shared": list(),
+            "detdata": [self.det_data],
+            "intervals": list(),
+        }
+        if self.shared_flags is not None:
+            req["shared"].append(self.shared_flags)
+        if self.det_flags is not None:
+            req["detdata"].append(self.det_flags)
+        if self.view is not None:
+            req["intervals"].append(self.view)
+        return req
+
+    def _provides(self):
+        prov = {
+            "meta": [self.noise_model],
```

Comment:
I don't think this operator provides a noise model- maybe a copy/paste residual?

Reply:
Thanks!